scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpression.scala

```diff
@@ -92,6 +104,37 @@ class SpelExpression(
 
   override val language: Language = flavour.languageId
 
+  def templateSubexpressions: Option[List[SpelTemplateSubexpression]] = {
+    def createTemplatedExpression(expression: org.springframework.expression.spel.standard.SpelExpression) = {
+      val parsedTemplateExpr = ParsedSpelExpression(expression.getExpressionString, parsed.parser, expression)
+      val compiledExpr = new SpelExpression(
+        parsedTemplateExpr,
+        typing.Typed[String],
+        Standard,
+        evaluationContextPreparer
+      )
+      TemplatedExpression(compiledExpr)
+    }
+    flavour.languageId match {
+      case Language.SpelTemplate =>
+        Some(parsed.parsed match {
+          case compositeExpr: CompositeStringExpression =>
+            compositeExpr.getExpressions.toList.map {
+              case lit: LiteralExpression => NonTemplatedValue(lit.getExpressionString)
```

Comment:
Can we leverage recurrence mechanism to flatten this?

Reply:
good idea, done :+1: